lib/kernel/doc/guides/socket_usage.md

```diff
@@ -127,6 +127,234 @@ And finally, its possible to override the global default when creating a socket
 the attribute `use_registry` (boolean) in the their `Opts` argument (which
 effects _that_ specific socket).
 
+
+## Example
+
+This example is intended to show how to create a simple (echo) server
+(and client).
+
+```erlang
+-module(example).
+
+-export([client/2, client/3]).
+-export([server/0, server/1, server/2]).
+
+
+%% ======================================================================
+
+%% === Client ===
+
+client(#{family := Family} = ServerSockAddr, Msg)
+  when is_list(Msg) orelse is_binary(Msg) ->
+    {ok, Sock} = socket:open(Family, stream, default),
+    ok         = socket:connect(Sock, ServerSockAddr),
+    client_exchange(Sock, Msg);
+
+client(ServerPort, Msg)
+  when is_integer(ServerPort) andalso (ServerPort > 0) ->
+    Family   = inet, % Default
+    Addr     = get_local_addr(Family), % Pick an address
+    SockAddr = #{family => Family,
+		 addr   => Addr,
+		 port   => ServerPort},
+    client(SockAddr, Msg).
+
+client(ServerPort, ServerAddr, Msg)
+  when is_integer(ServerPort) andalso (ServerPort > 0) andalso
+       is_tuple(ServerAddr) ->
+    Family   = which_family(ServerAddr),
+    SockAddr = #{family => Family,
+		 addr   => ServerAddr,
+		 port   => ServerPort},
+    client(SockAddr, Msg).
+
+%% Send the message to the (echo) server and wait for the echo to come back.
+client_exchange(Sock, Msg) when is_list(Msg) ->
+    client_exchange(Sock, list_to_binary(Msg));
```

Comment:
`socket:send/3` accepts an `iodata()` so converting to a `binary()` may be a bad example.